index.d.ts

```diff
@@ -26,13 +26,13 @@ export function getAlpha3Codes(): { [alpha3Key: string]: string };
  * Returns object map where key is numeric code and value is alpha 2 code
  */
 export function getNumericCodes(): { [numericKey: string]: string };
-export function getName(alpha2orAlpha3orNumeric: string | number, lang: string): string;
-export function getNames(lang: string): LocalizedCountryNames;
-export function toAlpha3(alpha2orNumeric: number | string): string;
-export function toAlpha2(alpha3orNumeric: number | string): string;
-export function getAlpha2Code(name: string, lang: string): string;
-export function getSimpleAlpha2Code(name: string, lang: string): string;
-export function getAlpha3Code(name: string, lang: string): string;
-export function getSimpleAlpha3Code(name: string, lang: string): string;
+export function getName(alpha2orAlpha3orNumeric: string | number, lang: string): string | undefined;
+export function getNames(lang: string): LocalizedCountryNames | undefined;
```

Comment:
getNames does not return undefined. It returns an empty object?

Reply:
You're totally right. 

```js
exports.getNames = function (lang) {
  var d = registeredLocales[lang.toLowerCase()];
  if (d === undefined) {
    return {};
  }
  return d;
};
```

Well, we can't return `{}` and `object` types here, as there will be a lot of cumbersome service code. Though we can return `Partial<LocalizedCountryNames>`, so the type of `getNames('en')['ru']` will be `string | string[] | undefined`. The other option is `lang` specification: for predefined `lang` values return `LocalizedCountryNames`, for the other return `Partial<LocalizedCountryNames>`.

At this step, I would prefer just `Partial<LocalizedCountryNames>`.

Reply:
If that makes sense for a TypeScript dev I'm fine with it :) I'm not using TypeScript so I have no opinion.

Reply:
@ibarrae what would you prefer?

Reply:
Ok, I will implement `Partial<...>`.

Reply:
I don't see a problem with using `{}`. You just need to check if a key is present, for instance

```
const names = getNames("en");
if ("US" in names) {
  // you can use it here
}
```